lib/resty/limit/count.md

```diff
@@ -0,0 +1,208 @@
+Name
+====
+
+resty.limit.count - Lua module for limiting request counts for OpenResty/ngx_lua.
+
+Table of Contents
+=================
+
+* [Name](#name)
+* [Synopsis](#synopsis)
+* [Description](#description)
+* [Methods](#methods)
+    * [new](#new)
+    * [incoming](#incoming)
+* [Installation](#installation)
+* [Author](#author)
+* [See Also](#see-also)
+
+Synopsis
+========
+
+```nginx
+http {
+    lua_shared_dict my_limit_count_store 100m;
+
+    init_by_lua_block {
+        require "resty.core"
+    }
+
+    server {
+        location / {
+            access_by_lua_block {
+                local limit_count = require "resty.limit.count"
+
+                -- rate: 5000 requests per 3600s
+                local lim, err = limit_count.new("my_limit_count_store", 5000, 3600)
+                if not lim then
+                    ngx.log(ngx.ERR, "failed to instantiate a resty.limit.count object: ", err)
+                    return ngx.exit(500)
+                end
+
+                -- use the Authorization header as the limiting key
+                local key = ngx.req.get_headers()["Authorization"] or "public"
+                local delay, err = lim:incoming(key, true)
+
+                if not delay then
+                    if err == "rejected" then
+                        ngx.header["X-RateLimit-Limit"] = "5000"
+                        ngx.header["X-RateLimit-Remaining"] = 0
```

Comment:
Hard-coded remaining?

Reply:
```
local delay, err = lim:incoming(key, true)
```
returns `nil, "rejected"` here,  and `remaining` must be 0 in this situation.